lib/oelite/cmd/show_provided.py

```diff
@@ -0,0 +1,185 @@
+    else:
+        logging.getLogger().setLevel(logging.INFO)
+
+INITIAL_OE_IMPORTS = "sys os time"
+
+# One must describe how to get the version of each util. Fortunately,
+# many utilities support a simple "--version" flag, and for most, the
+# command to call is the same as what's in the ASSUME_PROVIDED (an
+# exception is native:pkgconfig, where the command is pkg-config). So
+# we don't need to be very verbose.
+#
+# extract: a regexp string which is matched against the output from
+# the command. Should contain a single capture group, from which we'll
+# get the version string
+#
+# ret: Some utilities exit with a non-zero value when asked to print
+# their version (sometimes because they don't actually explicitly
+# support --version or any other flag, but they display their version
+# nevertheless as a side effect of telling the user he did something
+# wrong...). This can be set to an integer or list of integers of
+# expected return values.
+data = dict()
+data["native:binutils"] = dict(cmd = "ld")
+data["native:coreutils"] = dict(cmd = "ls")
+data["native:mercurial"] = dict(cmd = "hg")
+data["native:mtd-utils-mkfs-jffs2"] = dict(cmd = "mkfs.jffs2", ret = 255)
+data["native:perl"] = dict(extract = r"\(v([0-9][0-9.a-zA-Z_-]+)\)")
+data["native:pkgconfig"] = dict(cmd = "pkg-config")
+data["native:python-runtime"] = dict(cmd = "python")
+data["native:texinfo"] = dict(cmd = "texindex")
+data["native:unzip"] = dict(args = "-v")
+data["native:util-linux"] = dict(cmd = "getopt")
+
+def fill_defaults(data, assume_provided):
+    # Provide an empty dict for those where we don't have any
+    # requirements. That makes it easy for us to (try to) run the
+    # default "get the version provided", possibly fixing up those
+    # where that fails.
+    for name in assume_provided:
+        if name in data:
+            continue
+        # Libraries would need to be handled in some other way...
+        if name.startswith("native:lib"):
+            continue
+        data[name] = dict()
+    for name in data:
+        param = data[name]
+        if not "cmd" in param:
+            if ":" in name:
+                param["cmd"] = name.split(":", 1)[1]
+            else:
+                param["cmd"] = name
+
+        if not "args" in param:
+            param["args"] = "--version"
+        if isinstance(param["args"], basestring):
+            param["args"] = [param["args"]]
+
+        if not "extract" in param:
+            param["extract"] = r"\b([0-9][0-9.a-zA-Z_-]+)\b"
+
+        if not "ret" in param:
+            param["ret"] = 0
+        if (isinstance(param["ret"], int)):
+            param["ret"] = [param["ret"]]
+
+def fill_versions(data, key, versions):
+    for util in versions:
+        u, v = util.split("_", 1)
+        if u in data:
+            data[u][key] = v
+
+def get_provided_version(p):
+    param = data.get(p)
+    if not param:
+        return None
+
+    args = [param["cmd"]] + param["args"]
+    cmdstring = " ".join(args)
```

Comment:
Why are you flattening args?  subprocess.Popen() is capable of taken args as a sequence, and in case you have args with whitespace, you effectively split the arguments this way.